src/TAMSEquationSystem.C

```diff
@@ -0,0 +1,615 @@
+/*------------------------------------------------------------------------*/
+/*  Copyright 2014 Sandia Corporation.                                    */
+/*  This software is released under the license detailed                  */
+/*  in the file, LICENSE, which is located in the top-level Nalu          */
+/*  directory structure                                                   */
+/*------------------------------------------------------------------------*/
+
+#include <AlgorithmDriver.h>
+#include <ComputeTAMSAvgMdotEdgeAlgorithm.h>
+#include <ComputeTAMSAvgMdotElemAlgorithm.h>
+#include <ComputeMetricTensorNodeAlgorithm.h>
+#include <ComputeSSTTAMSAveragesNodeAlgorithm.h>
+#include <EquationSystem.h>
+#include <EquationSystems.h>
+#include <Enums.h>
+#include <FieldFunctions.h>
+#include <NaluEnv.h>
+#include <NaluParsing.h>
+#include <Realm.h>
+#include <Realms.h>
+#include <Simulation.h>
+#include <SolutionOptions.h>
+#include <TAMSEquationSystem.h>
+#include <TimeIntegrator.h>
+#include <TurbViscSSTTAMSAlgorithm.h>
+
+#include <SolverAlgorithmDriver.h>
+
+// template for supp algs
+#include <AlgTraits.h>
+#include <kernel/KernelBuilder.h>
+#include <kernel/KernelBuilderLog.h>
+
+// stk_util
+#include <stk_util/parallel/Parallel.hpp>
+
+// stk_mesh/base/fem
+#include <stk_mesh/base/BulkData.hpp>
+#include <stk_mesh/base/Field.hpp>
+//#include <stk_mesh/base/FieldParallel.hpp>
+//#include <stk_mesh/base/GetBuckets.hpp>
+//#include <stk_mesh/base/GetEntities.hpp>
+//#include <stk_mesh/base/CoordinateSystems.hpp>
+#include <stk_mesh/base/MetaData.hpp>
+
+// stk_io
+#include <stk_io/IossBridge.hpp>
+
+// stk_topo
+#include <stk_topology/topology.hpp>
+
+// stk_util
+#include <stk_util/parallel/ParallelReduce.hpp>
+
+namespace sierra {
+namespace nalu {
+
```

Comment:
This isn't an equation system.  Just bundle together the algorithms you want to run and the field registrations in a class and add some logic e.g. in ShearStressTransport or maybe even TimeIntegrator to get them to execute at the right time.  There's no need to try to fit it into an equation system unless you're planning on solving a PDE as part of the model soon.

Reply:
So there is a possibility we would want to solve a projection of the forcing field onto a divergence free space as part of the TAMS implementation.  For now we aren't pursuing that but initially that was an option and could be one going forward.

Let us know if you think it is best to merge it into ShearStressTransport now and split it off down the road if we go that route or if you think it would be okay to leave it as an Equation System for now and merge it with ShearStress if the projection option is ruled out more permanently?

Reply:
It took me a while to understand this too. Can the TAMS equation system just inherit the ShearStressTransportEquationSystem?

Reply:
@gantech you can make a "TAMSManager" (or "TAMSManagerSST" or somerhing) class that registers fields and executes some algorithms instead of having it as an equation system.  An instance of that class could be a member of the ShearStressTransport equations or Realm or depending on exactly when you want the algorithms to be executed in the overall sequence.

@jamelvin The second option of keeping it as it is for now and then looking to merge it later if the projection isn't really unnecessary sounds good to me.